packages/api/src/cors.ts

```diff
@@ -0,0 +1,73 @@
+/* eslint-env serviceworker */
+import type { RouteHandler } from 'itty-router'
+
+/**
+ * Handler for CORS OPTIONS requests.
+ */
+export function corsOptions (request: Request): Response {
+  const headers = request.headers
+  // Make sure the necessary headers are present for this to be a valid pre-flight request
+  if (
+    headers.get('Origin') !== null &&
+    headers.get('Access-Control-Request-Method') !== null &&
+    headers.get('Access-Control-Request-Headers') !== null
+  ) {
+    function headerOrDefault (key: string, fallback: string): string {
+      const value = headers.get(key)
+      if (value !== '' && value !== null) {
+        return value
+      }
+      return fallback
+    }
+    // Handle CORS pre-flight request.
+    /** @type {Record<string, string>} */
+    const respHeaders = {
+      'Content-Length': '0',
+      'Access-Control-Allow-Origin': headerOrDefault('origin', '*'),
+      'Access-Control-Allow-Methods': 'GET,POST,DELETE,OPTIONS',
```

Comment:
I don't think we have any DELETE endpoints? So could omit this for now tooo

Reply:
Again, a good point.